docs/dev/resources/networking.md

```diff
@@ -0,0 +1,57 @@
+This is a bit more niche, as our unclassified notebooks already have connection to the internet, but this can come up when we have a Protected B certified application like seen in https://github.com/StatCan/aaw-private/issues/125 where it's accessible to the internet, but unaccecssible from our Protected B workloads.
+
+To handle this we want to declare a [firewall_policy_rule_collection_group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall_policy_rule_collection_group) in our terraform files. 
+
+Regarding certain fields, the rule collection group type we chose was [`application rule`](https://learn.microsoft.com/en-us/azure/firewall/policy-rule-sets#application-rules) as we are filtering based on FQDNs.
+
+The `source address` is the subnet address of our protected-b node pool. 
+
+-------------
+### Allow Connection to internal application in an already peered network
+An example of this can be seen in https://github.com/StatCan/aaw-private/issues/127 
+
+- __Important!__: This solution does not work fully just yet, because of the [DNS pitfall](#dns-pitfalls-to-internal-application-connecting) that we encountered so this will need to be updated.
+
+Like above, this also requires a [firewall_policy_rule_collection_group](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/firewall_policy_rule_collection_group) in our terraform files. 
+Regarding certain fields, the rule collection group type we chose was [`network rule`](https://learn.microsoft.com/en-us/azure/firewall/policy-rule-sets#network-rules) as in this case we are working with IP/subnet addresses
+In the example, the `source_addresses`s were two entries, one for our unclassified node pool subnet and the other for our protected b node pool subnet.
+
+Additionally, it will require an [azurem_route](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/route) to manage the [routing (azure docs)](https://learn.microsoft.com/en-us/azure/virtual-network/virtual-networks-udr-overview#user-defined). 
+The `address_prefix`is the destination to which the route applies (in this case the subnet that the database resides in).
+Then for the `next_hop_type` we chose `VirtualAppliance` as we have to get to the cloud main firewall first. 
+The `next_hop_in_ip_address` is the cloud main firewall ip.
+
+
+The other aspect to this is that we also need flow to be opened up on the other side, as in the flow has to be allowed from the firewall to the actual application. This is out of the AAW's hands so an SR must be submitted for this to be done.
+
+>  If the application is in a network that is not already peered, in addition to the route you will neeed to create an [`azurem_virtual_network_peering`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_network_peering.html) resource.
+
+-----------
+### DNS Pitfalls to internal application connecting

```

Comment:
Maybe add some general notes about DNS before the pitfalls:
- for accessing internal resources by FQDN, a private DNS zone is required to map the IP address to the service name
- DNS rules are needed for name resolution and are technically separate from the firewall rules needed to allow access - both are required
- some internal services built on azure SaaS platforms may already have internet accessible DNS configured, so check if this is necessary at all

The issue described is simply caused by two services under the same domain where one service already has public DNS and the other does not, the private dns zone captures all requests for the domain!

Reply:
Added this bar for bar note for note